src/AppInstallerCommonCore/AppInstallerLogging.cpp

```diff
@@ -8,6 +8,8 @@
 #include "Public/AppInstallerDateTime.h"
 #include "Public/AppInstallerRuntime.h"
 
+thread_local AppInstaller::Logging::ThreadGlobals* t_pThreadGlobals = nullptr;
```

Comment:
The thread_local should be inside a function.

I'm not strictly set on this, but I feel like it provides defense against bad behavior.